x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/format_selector.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiFieldNumber, EuiComboBox } from '@elastic/eui';
+import { IndexPattern } from '../types';
+import { IndexPatternColumn } from '../indexpattern';
+import { IndexPatternDimensionPanelProps } from './dimension_panel';
+
+const supportedFormats: Record<string, { title: string }> = {
+  number: {
+    title: i18n.translate('xpack.lens.indexPattern.numberFormatLabel', {
+      defaultMessage: 'Number',
+    }),
+  },
+  percent: {
+    title: i18n.translate('xpack.lens.indexPattern.percentFormatLabel', {
+      defaultMessage: 'Percent',
+    }),
+  },
+  bytes: {
+    title: i18n.translate('xpack.lens.indexPattern.bytesFormatLabel', {
+      defaultMessage: 'Bytes (1024)',
+    }),
+  },
+};
+
+type FormatSelectorProps = Pick<IndexPatternDimensionPanelProps, 'data'> & {
+  selectedColumn: IndexPatternColumn;
```

Comment:
I think we already should limit here down to columns, that support accepting a format (another reason to extract that part into it's own type). That way we would not need to cast in line #118 further down (at least we could it move to the popover_editor, where it's closer to the check it belongs to), and basically have this component really type safe (at the moment you could actually pass in any column you want, and we'd still try to get the format params from it).